### PR TITLE
API docs: Add custom target to build for GH pages.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

**Comment by nafur** [comment]
I'd rather avoid moving stuff around so much. I just followed you down the rabbit hole of the issues with the `_static` folder. Apparently this was needed because jekyll (which gh pages uses internally) ignores folders with underscores. However, you can also put a `.nojekyll` file and disable jekyll altogether. (See https://github.com/sphinx-doc/sphinx/issues/2202 and https://www.sphinx-doc.org/en/master/usage/extensions/githubpages.html).
It suffices to add `'sphinx.ext.githubpages'` to the `extensions` list.

File: docs/CMakeLists.txt [hunk 1]
```diff
@@ -45,9 +45,29 @@ set(SPHINX_OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/sphinx)
...
+                    ${SPHINX_OUTPUT_DIR} ${SPHINX_GH_OUTPUT_DIR}
+                  COMMAND ${CMAKE_COMMAND} -E remove_directory
+                    ${SPHINX_GH_OUTPUT_DIR}/_sources
+                  COMMAND ${CMAKE_COMMAND} -E remove
+                    ${SPHINX_GH_OUTPUT_DIR}/objects.inv
+                  COMMAND ${CMAKE_COMMAND} -E rename
+                    ${SPHINX_GH_OUTPUT_DIR}/_static
+                    ${SPHINX_GH_OUTPUT_DIR}/static
+                  COMMAND find ${SPHINX_GH_OUTPUT_DIR} -type f |
+                    xargs sed -i'orig' 's/_static/static/'
```
Comment (nafur) [inline docs/CMakeLists.txt:70]:
```suggestion
```
Instead: add `'sphinx.ext.githubpages'` to `extensions` in `conf.py`.

File: docs/CMakeLists.txt [hunk 2]
```diff
@@ -45,9 +45,29 @@ set(SPHINX_OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/sphinx)
...
+                  COMMAND find ${SPHINX_GH_OUTPUT_DIR} -name '*orig' -delete
```
Comment (nafur) [inline docs/CMakeLists.txt:71]: I don't find any such files in my local build. However, I find `.buildinfo` and `.doctrees`

Comment (nafur) [inline docs/CMakeLists.txt:71]:
```suggestion
```
Instead: add `'sphinx.ext.githubpages'` to `extensions` in `conf.py`.

**Comment by mpreiner** [comment]
> I'd rather avoid moving stuff around so much. I just followed you down the rabbit hole of the issues with the `_static` folder. Apparently this was needed because jekyll (which gh pages uses internally) ignores folders with underscores. However, you can also put a `.nojekyll` file and disable jekyll altogether. (See [sphinx-doc/sphinx#2202](https://github.com/sphinx-doc/sphinx/issues/2202) and https://www.sphinx-doc.org/en/master/usage/extensions/githubpages.html).
> It suffices to add `'sphinx.ext.githubpages'` to the `extensions` list.

`.nojekyll` only works if it's in the root directory of the repository. Since the documentation will be part of the CVC5 website it will be in a subdirectory of the jekyll page.

**Comment by nafur** [comment]
> > I'd rather avoid moving stuff around so much. I just followed you down the rabbit hole of the issues with the `_static` folder. Apparently this was needed because jekyll (which gh pages uses internally) ignores folders with underscores. However, you can also put a `.nojekyll` file and disable jekyll altogether. (See [sphinx-doc/sphinx#2202](https://github.com/sphinx-doc/sphinx/issues/2202) and https://www.sphinx-doc.org/en/master/usage/extensions/githubpages.html).
> > It suffices to add `'sphinx.ext.githubpages'` to the `extensions` list.
> 
> `.nojekyll` only works if it's in the root directory of the repository. Since the documentation will be part of the CVC5 website it will be in a subdirectory of the jekyll page.

I see.

**Comment by aniemetz** [comment]
I just tested the extension and the only thing it seems to do is create the `.nojekyll` file, so this is unfortunately no solution for us.

**Review by nafur — APPROVED** [review]
LGTM. I assume there will be another PR that uses this in a CI job?

**Comment by aniemetz** [comment]
Yes, this is postponed for when we try to automatize the process to push to the website.

**Pull request merged**